tests/org.eclipse.passage.loc.billing.core.tests/src/org/eclipse/passage/loc/billing/core/tests/UserLicensesTest.java

```diff
@@ -14,75 +14,93 @@
 
 		assertEquals(Stream.concat(licenses1.stream(), licenses2.stream()).collect(Collectors.toList()),
 				service.getAllLicenses());
 	}
 
 	@Test
 	public void licensesForProduct() {
-		String product = "product"; //$NON-NLS-1$
-		FakeLicenseDescriptor license1 = new FakeLicenseDescriptor(product);
-		FakeLicenseDescriptor license2 = new FakeLicenseDescriptor(product);
+		String product = provider.randomString();
+		List<UserLicenseDescriptor> licenses1 = getLicenses(getLicense(product), getLicense(provider.randomString()));
+		List<UserLicenseDescriptor> licenses2 = getLicenses(getLicense(product), getLicense(provider.randomString()));
 
-		List<UserLicenseDescriptor> licenses1 = new ArrayList<UserLicenseDescriptor>();
-		licenses1.add(license1);
-		licenses1.add(new FakeLicenseDescriptor("bar")); //$NON-NLS-1$
+		UserLicenses service = getService(getUser(licenses1), getUser(licenses2));
+
+		assertEquals(Arrays.asList(getLicense(product), getLicense(product)), service.getLicensesForProduct(product));
+	}
 
-		List<UserLicenseDescriptor> licenses2 = new ArrayList<UserLicenseDescriptor>();
-		licenses2.add(new FakeLicenseDescriptor("baz")); //$NON-NLS-1$
-		licenses2.add(license2);
+	@Test
+	public void licensesForProductVersion() {
+		String product = provider.randomString();
+		String version = provider.randomString();
 
-		UserLicenses service = new UserLicenses(
-				Arrays.asList(new FakeUser[] { new FakeUser(licenses1), new FakeUser(licenses2) }));
+		UserLicenses service = getService(
+				getUser(getLicenses(getLicense(product, version), getLicense(provider.randomString()))),
+				getUser(getLicenses(getLicense(product), getLicense(provider.randomString()))));
 
-		assertEquals(Arrays.asList(license1, license2), service.getLicensesForProduct(product));
+		assertEquals(Arrays.asList(getLicense(product, version)),
+				service.getLicensesForProductVersion(product, version));
 	}
 
 	@Test
-	public void licensesForProductVersion() {
-		String product = "product"; //$NON-NLS-1$
-		String version = "version"; //$NON-NLS-1$
+	public void countSameLicenses() {
+		UserLicenseDescriptor license1 = getLicense(provider.randomString(), provider.randomString());
+		UserLicenseDescriptor license2 = getLicense(provider.randomString(), provider.randomString());
+		UserLicenseDescriptor license3 = getLicense(provider.randomString(), provider.randomString());
+
+		UserLicenses service = getService(getUser(getLicenses(license1, license3)),
+				getUser(getLicenses(license2, license3)), getUser(getLicenses(license1, license2, license3)));
 
-		FakeLicenseDescriptor license1 = new FakeLicenseDescriptor(product, version);
-		FakeLicenseDescriptor license2 = new FakeLicenseDescriptor(product); // default version is empty string
+		assertEquals(Arrays.asList(2, 2, 3), getNumbers(service, license1, license2, license3));
+	}
+
+	private List<Integer> getNumbers(UserLicenses service, UserLicenseDescriptor... descriptors) {
```

Comment:
These six are [queries](https://www.yegor256.com/2018/08/22/builders-and-manipulators.html) (builders). Let them be named with a noun after the thing they query/build.  Just like you did in your `StringProvider` above  (see `randomString`):
```java
private UserLicenseDescriptor license(String product, String version)
```

Thus your client code will be more readable: 
```java
service = service(user(licenses(license1, license3))
```